src/transaction/deserialize.c

```diff
@@ -103,18 +108,18 @@ parser_status_e tx_variant_deserialize(buffer_t *buf, transaction_t *tx) {
 
     uint8_t *prefix;
     // read hashed prefix bytes
-    if (!bcs_read_ptr_to_fixed_bytes(buf, &prefix, TX_HASHED_PREFIX_LEN)) {
-        return HASHED_PREFIX_READ_ERROR;
-    }
-
-    if (memcmp(prefix, PREFIX_RAW_TX_WITH_DATA_HASHED, TX_HASHED_PREFIX_LEN) == 0) {
-        tx->tx_variant = TX_RAW_WITH_DATA;
-        return PARSING_OK;
-    }
+    if (bcs_read_ptr_to_fixed_bytes(buf, &prefix, TX_HASHED_PREFIX_LEN)) {
+        if (memcmp(prefix, PREFIX_RAW_TX_WITH_DATA_HASHED, TX_HASHED_PREFIX_LEN) == 0) {
+            tx->tx_variant = TX_RAW_WITH_DATA;
+            return PARSING_OK;
+        }
 
-    if (memcmp(prefix, PREFIX_RAW_TX_HASHED, TX_HASHED_PREFIX_LEN) == 0) {
-        tx->tx_variant = TX_RAW;
-        return PARSING_OK;
+        if (memcmp(prefix, PREFIX_RAW_TX_HASHED, TX_HASHED_PREFIX_LEN) == 0) {
+            tx->tx_variant = TX_RAW;
+            return PARSING_OK;
+        }
+    } else {
+        status = HASHED_PREFIX_READ_ERROR;
     }
 
     if (transaction_utils_check_encoding(buf->ptr, buf->size)) {
```

Comment:
`transaction_utils_check_encoding` will still return `true` if the buf contains a NUL character within the message:
If the transaction buffer contains `"ABC\0EFG\0"` will be accepted as a message; but in the display, only "ABC" will be presented to the user. I would suggest to consider as a message, _a buffer containing non-null ascii characters for N-1 first bytes, and a training NUL char_.